packages/sdk/react-client/src/echo/SpaceContext.tsx

```diff
@@ -34,7 +40,9 @@ export const SpaceProvider = ({ initialSpaceKey, onSpaceChange, children }: Spac
   });
 
   useEffect(() => {
-    onSpaceChange?.(spaceKey);
+    if (spaceKey) {
+      onSpaceChange?.(spaceKey);
+    }
```

Comment:
Why should deselection not fire this event?

Reply:
it probably should but i had some downstream issues -- e.g., that this would trigger creating a new space.
the app route/space selection/auto-creation needs a bit of a think.